### PR TITLE
Fix for invalid view deployment with no user access

### PR DESCRIPTION
**Pull request by kushalmalani**
## Description

> Describe the purpose of this pull request.

Fix for invalid view deployment with no user access

## 🎟 Issue(s)

Resolves astronomer/issues#2130

## 🧪 Functional Testing

> List the functional testing steps to confirm this feature or fix.

1. Create Workspace and Deployment
2. Invite User to workspace but not to deployment
3. Try to navigate to the deployment url by hitting the url directly from browser
4. Non System Admin User shouldnt be able to view deployment since he was not added to it

## 📸 Screenshots

Screen Shot 2020-11-17 at 7 11 57 PM

## 📋 Checklist

- [x] Rebased from the main (or release if patching) branch (before testing)
- [x] Ran `yarn lint && yarn test` before taking out of draft
- [x] Added/updated applicable tests
- [x] Tested against [Astro-UI](https://github.com/astronomer/astro-ui/) and [Astro-CLI](https://github.com/astronomer/astro-cli/) clients.
- [ ] Communicated to/tagged owners of respective clients potentially impacted by these changes.
- [ ] Updated any related [documentation](https://github.com/astronomer/docs/)

### CODE REVIEW

File: src/lib/workspaces/index.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import { compact, find, get } from "lodash";
+import { SYSTEM_ADMIN } from "constants";
+
+/*
+ * Build the deployments query based on args.
+ * @param {Object} args The graphql arguments.
+ * @param {Object} ctx The graphql context.
+ * @return {Object} The deployments query.
+ */
+export function deploymentsQuery(args, ctx) {
```
Comment (kushalmalani) [inline src/lib/workspaces/index.js:10]: Move this funciton to a library since this function was being reused at multiple places.

**Comment by adam2k** [comment]
The code looks good to me.  I need to functional test to confirm.  I'm working on some Astro Dev stuff at the moment, so it'll be a minute.

**Comment by kushalmalani** [comment]
> The code looks good to me. I need to functional test to confirm. I'm working on some Astro CLI Airflow Dev stuff at the moment, so it'll be a minute.

Sure. I have listed the steps above and in the ticket. Let me know if its not clear.

**Review by adam2k — APPROVED** [review]
LGTM!  I did functional testing and I wasn't able to see the deployment with the new user until I was added to the deployment by the admin.

**Pull request merged**